### PR TITLE
Update AudioTrack layouts as per new designs

### PR DESCRIPTION
**Pull request by dhruvkb**
## Description

This PR updates the `AudioTrack` layouts, including subcomponents like `Waveform` to match the new mockups.

### Changes

- Made `Waveform` component extensible to seamlessly fit in different layouts
  - Fixed the `Waveform` border to render edge to edge with other components
  - Background can now be changed
  - Bits of UI such as duration and timestamps can be toggled
- Added new layouts and sizes
  - Added `Box` layout in 3 sizes (absorbing and extending #230)
  - Added S size for `Full` layout (not 100% ready because #510)
- Addressed focus and hover states for all `AudioTrack` layouts
- Updated documentation
  - Now clearly indicates default layouts and sizes
  - Enabled Storybook Viewport addon and configured mobile views to render in mobile viewport 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -45,21 +44,25 @@ module.exports = {
...
-      'dark-charcoal-06': '#f3f2f2',
-      'dark-charcoal-10': '#eae9ea',
-      'dark-charcoal-20': '#d6d4d5',
-      'dark-charcoal-30': '#c1bec0',
-      'dark-charcoal-40': '#aca9ab',
-      'dark-charcoal-50': '#989397',
-      'dark-charcoal-60': '#837d82',
-      'dark-charcoal-70': '#6e686d',
-      'dark-charcoal-80': '#595258',
+      'dark-charcoal': {
```
Comment (obulat) [inline tailwind.config.js:47]: This will break so many other components. We'll need to find/replace before merging this PR.

File: nuxt.config.js [hunk 2]
```diff
@@ -222,7 +223,7 @@ export default {
         name: '@storybook/addon-essentials',
         options: {
           backgrounds: false,
-          viewport: false,
+          viewport: true,
```
Comment (obulat) [inline nuxt.config.js:226]: Thank you for adding the viewports! I wanted to add them for Header :)

**Review by obulat — APPROVED** [review]
The component looks amazing!
One question about the license icon in the Box layout: is it supposed to be hidden and only shown on hover?

**Review by zackkrida — APPROVED** [review]
Really nice!

File: tailwind.config.js (hunk 1 above)
Comment (dhruvkb) [reply tailwind.config.js:47]: I already did the find-and-replace.

**Comment by dhruvkb** [comment]
@obulat the licenses appearing on hover was behaviour was confirmed by @panchovm.

**Pull request merged**